File: 2540-minimum-common-value/2540-minimum-common-value.cpp
class Solution {
public:
    int getCommon(vector<int>& nums1, vector<int>& nums2)
    {
         map<long long int ,long long int>mp;
        for(int i=0;i<nums1.size();i++)
        {
            mp[nums1[i]]++;
        }
        
          map<int , int>mp1;
        for(int i=0;i<nums2.size();i++)
        {
            mp1[nums2[i]]++;
        }
         for(auto it:mp)
         {
             if(mp1.find(it.first)!=mp1.end())
             {
                 return it.first;
             }
         }
        return -1;
        
        
    }
};

File: 2540-minimum-common-value/2540-minimum-common-value.ts
function getCommon(nums1: number[], nums2: number[]): number {
    let i = 0, j = 0

    while (i < nums1.length && j < nums2.length) {
        if (nums1[i] === nums2[j]) return nums1[i]
        nums1[i] < nums2[j] ? i++ : j++
    }

    return -1
};